SmartAgentConcepts.md

```diff
@@ -0,0 +1,129 @@
+# SignalFx Smart Agent Concepts
```

Comment:
Put this file at `docs/concepts.md`.  Docs should all go under `docs/`, except for the main README.md at the root.